core/tests/raw_swarm_simult.rs

```diff
@@ -209,11 +233,12 @@ fn raw_swarm_simultaneous_connect() {
                         }
                     }
 
-                    if swarm1_step == 3 && swarm2_step == 3 {
+                    // TODO: make sure that >= 5 is correct
+                    if swarm1_step + swarm2_step >= 5 {
```

Comment:
If a node dials, then receives an incoming connection, it will switch to connected and be only at step 2. Then there's no further event. Hence the ` >= 5`, so either must be `3` and the other can be `2`. However I think I should make this a bit stronger before we merge.